### PR TITLE
Merge ashlar

### PR DESCRIPTION
**Pull request by alliecreason**
Pull in ashlar with history from [old repo](https://github.com/ohsu-comp-bio/ashlar).

### CODE REVIEW

File: tools/ashlar/ashlar.xml [hunk 1]
```diff
@@ -0,0 +1,184 @@
+<tool id="ashlar" name="ASHLAR" version="@VERSION@.1" profile="17.09">
```
Comment (bgruening) [inline tools/ashlar/ashlar.xml:1]: I guess the profile version can be increased?
We recommend using TOOL_VERSION https://galaxy-iuc-standards.readthedocs.io/en/latest/best_practices/tool_xml.html#tool-versions 

If you stick to it we have a bot that can bump the requirement versions etc ...

File: tools/ashlar/ashlar.xml [hunk 2]
```diff
@@ -0,0 +1,184 @@
...
+        #end for
+        #for $ffp in $lffp:
+            ln -s "$ffp" "$clean($ffp,"ffp")" &&
+        #end for
+
+        @CMD_BEGIN@
+
+        ## Supply the raw images
+        #for $raw in $lraw:
+            "$raw"
```
Comment (bgruening) [inline tools/ashlar/ashlar.xml:38]: single quotes

File: tools/ashlar/ashlar.xml [hunk 3]
```diff
@@ -0,0 +1,184 @@
...
+        --filter-sigma $adv.filter_sigma
+        #end if
+
+        #if $adv.tile_size
+        --tile-size $adv.tile_size
+        #end if
+
+        --ffp
+        #for $ffp in $lffp:
+            "$clean($ffp,"ffp")"
```
Comment (bgruening) [inline tools/ashlar/ashlar.xml:64]: single-quotes

File: tools/ashlar/ashlar.xml [hunk 4]
```diff
@@ -0,0 +1,184 @@
...
+        #end if
+
+        #if $adv.flip_mosaic_y
+        --flip-mosaic-y
+        #end if
+
+        -f registered.ome.tif;
+
+        #if $upgrade.decide == "do_upgrade"
+        python3 ${__tool_directory__}/pyramid_upgrade.py
```
Comment (bgruening) [inline tools/ashlar/ashlar.xml:87]: single-quotes, essentially all paths and text params should be single-quoted

File: tools/ashlar/ashlar.xml [hunk 5]
```diff
@@ -0,0 +1,184 @@
...
+        registered.ome.tif
+
+        #if $upgrade.markers_file
+        -n `echo \$(cat $upgrade.markers_file | tail -n +2 | awk -F, '{print \$3}')`;
+        #end if
+        #end if
+    ]]></command>
+
+    <inputs>
+        <param name="lraw" type="data_collection" format="ome.tiff,tiff" collection_type="list" label="Raw Images"/>
```
Comment (bgruening) [inline tools/ashlar/ashlar.xml:97]: This tool can also work on a single file, isn't it? Why not using "type="data" multiple="true""?

File: tools/ashlar/ashlar.xml [hunk 6]
```diff
@@ -0,0 +1,184 @@
...
+            <param name="flip_mosaic_x" type="boolean" value="false" label="Flip output image horizontally"/>
+            <param name="flip_mosaic_y" type="boolean" value="false" label="Flip output image vertically"/>
+            <param name="pyramid" type="boolean" value="true" label="Write output as a single pyramidal TIFF"/>
+        </section>
+    </inputs>
+
+    <outputs>
+        <data format="ome.tiff" name="output" from_work_dir="registered.ome.tif" label="${tool.name} on ${on_string}"/>
+    </outputs>
+
```
Comment (bgruening) [inline tools/ashlar/ashlar.xml:130]: tests are missing

File: tools/ashlar/ashlar.xml [hunk 7]
```diff
@@ -0,0 +1,184 @@
...
+        #if $adv.pyramid
+        --pyramid
+        #end if
+
+        #if $adv.flip_mosaic_x
+        --flip-mosaic-x
+        #end if
+
+        #if $adv.flip_mosaic_y
```
Comment (bgruening) [inline tools/ashlar/ashlar.xml:80]: You can simplify booleans a lot. If you add the the param description below something like `truevalue="--flip-mosaic-y" falsevalue=""` you can write here just `$adv.flip_mosaic_y` and remove the entire "if"

File: tools/ashlar/meta.yaml [hunk 8]
```diff
@@ -0,0 +1,54 @@
+{% set name = "ashlar" %}
```
Comment (bgruening) [inline tools/ashlar/meta.yaml:1]: is this file needed here? I guess this should go to conda-forge or bioconda? I can help you with that, just ping me on the PR.

Comment (alliecreason) [reply tools/ashlar/meta.yaml]: bioconda/bioconda-recipes#35846

File: tools/ashlar/ashlar.xml (hunk 5 above)
Comment (bgruening) [reply tools/ashlar/ashlar.xml:84]: multiple=true mainly allows you to put in collections AND single datasets. If you want to disallow single datasets that is fine, just unusual.

File: tools/ashlar/meta.yaml (hunk 8 above)
Comment (bgruening) [reply tools/ashlar/meta.yaml]: on it

**Comment by bgruening** [comment]
@alliecreason you can test the package at https://github.com/bioconda/bioconda-recipes/pull/35876

**Comment by bgruening** [comment]
Lets use the Docker container for now. The old version has too many unresolved dependencies :(

File: tools/ashlar/macros.xml [hunk 9]
```diff
@@ -0,0 +1,21 @@
+<?xml version="1.0"?>
+<macros>
+    <xml name="requirements">
+        <requirements>
+            <requirement type="package" version="@TOOL_VERSION@">ashlar</requirement>
```
Comment (bgruening) [inline tools/ashlar/macros.xml:5]: if we go docker only we can remove the package line

File: tools/ashlar/macros.xml [hunk 10]
```diff
@@ -0,0 +1,21 @@
...
+            <container type="docker">labsyspharm/ashlar:@TOOL_VERSION@</container>
+        </requirements>
+    </xml>
+
+    <xml name="version_cmd">
+        <version_command>echo @TOOL_VERSION@</version_command>
+    </xml>
+    <xml name="citations">
+        <citations>
```
Comment (bgruening) [inline tools/ashlar/macros.xml:14]: no citation?

File: tools/ashlar/ashlar.xml [hunk 11]
```diff
@@ -0,0 +1,191 @@
...
+            </param>
+            <output name="output" ftype="ome.tiff">
+                <assert_contents>
+                    <has_size value="4000000" delta="1000000" />
+                </assert_contents>
+            </output>
+        </test>
+    </tests>
+    <help><![CDATA[
+Ashlar python package for microscopy registration, developed by HMS (repo: https://github.com/labsyspharm/ashlar)
```
Comment (bgruening) [inline tools/ashlar/ashlar.xml:139]: the help is not really useful for Galaxy users I think.

File: tools/ashlar/ashlar.xml [hunk 12]
```diff
@@ -0,0 +1,191 @@
...
+        $adv.flip_mosaic_x
+        $adv.flip_mosaic_y
+
+        -f registered.ome.tif;
+
+        #if $upgrade.decide == "do_upgrade"
+        python3 '${__tool_directory__}/pyramid_upgrade.py'
+        registered.ome.tif
+
+        #if $upgrade.markers_file
```
Comment (bgruening) [inline tools/ashlar/ashlar.xml:86]: indentation could be improved

**Review by bgruening — COMMENTED** [review]
Just some tiny details, overall this looks good to me!

File: tools/ashlar/macros.xml (hunk 9 above)
Comment (alliecreason) [reply tools/ashlar/macros.xml:5]: Keeping package will make it compatible for users that want to stick with conda.

**Pull request merged**